libraries/WiFi/src/WiFiSTA.cpp

```diff
@@ -391,6 +391,31 @@ bool WiFiSTAClass::disconnect(bool wifioff, bool eraseap)
     return false;
 }
 
+/**
+ * Disconnect from the network.
+ * @param wifioff `true` to turn the Wi-Fi radio off.
+ * @param eraseap `true` to erase the AP configuration from the NVS memory.
+ * @param timeoutLength timeout to wait for status change
+ * @return `true` when successful.
+ */
+bool WiFiSTAClass::disconnect(bool wifioff, bool eraseap, unsigned long timeoutLength)
+{
+    if (!disconnectAsync(wifioff, eraseap)) {
+        return false;
+    }
+    if (!timeoutLength) {
+        return true;
+    }
+    unsigned long start = millis();
```

Comment:
Please make this const.

Reply:
Well it is more like I trust the compiler to make further optimizations when I tell it some values are not supposed to change.
And when reading `const` in the code, I (as a person) can read code much faster as I don't have to keep track of it in my mind to see where it might be causing issues.
So it is just a habbit of me to use const whenever possible.